src/main/java/minicraft/screen/QuestsDisplay.java

```diff
@@ -0,0 +1,342 @@
+package minicraft.screen;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import minicraft.core.Game;
+import minicraft.core.io.InputHandler;
+import minicraft.core.io.Localization;
+import minicraft.gfx.Color;
+import minicraft.gfx.Point;
+import minicraft.gfx.Screen;
+import minicraft.saveload.Load;
+import minicraft.screen.entry.ListEntry;
+import minicraft.screen.entry.SelectEntry;
+import minicraft.screen.entry.StringEntry;
+import minicraft.util.Quest;
+
+public class QuestsDisplay extends Display {
+	private static HashMap<String, Quest> quests = new HashMap<>();
+	private static ArrayList<Quest> unlockedQuests = new ArrayList<>();
+	private static ArrayList<Quest> completeQuest = new ArrayList<>();
+	private static HashMap<String, QuestStatus> questStatus = new HashMap<>();
+	private static ArrayList<String> initiallyUnlocked = new ArrayList<>();
+
+	private Menu.Builder builder;
+	private SelectEntry[][] questEntries;
+	private int selectedEntry;
+	private HashMap<String, Quest> questWithLocalized;
+	private boolean entrySelected = false;
+
+	static {
+		// TODO Implementing tutorials and story mode
+		try {
+			JSONArray json = new JSONArray(String.join("", Load.loadFile("/resources/quests.json")));
+			for (int i = 0; i < json.length(); i++) {
+				JSONObject obj = json.getJSONObject(i);
+				String id = obj.getString("id");
+				boolean unlocked = obj.optBoolean("unlocked", false); // Is unlocked initially
+				JSONArray unlocksJson = obj.getJSONArray("unlocks");
+
+				String[] unlocks = new String[unlocksJson.length()];
+				for (int j = 0; j < unlocksJson.length(); j++) {
+					unlocks[j] = unlocksJson.getString(j);
+				}
+
+				if (unlocked)
+					initiallyUnlocked.add(id);
+
+				quests.put(id, unlocked? new Quest(id, obj.getString("desc"), unlocked, unlocks): new Quest(id, obj.getString("desc"), unlocks));
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
```

Comment:
The formatting here is nearly perfect. Everything is perfect other than some operators not having space on both sides.

Reply:
You still haven't fixed the changes I requested.

Reply:
Ok